### PR TITLE
Reducer and action

### PR DESCRIPTION
**Pull request by ankitt26**
In this branch,I did -

- Add Redux Toolkit
- Configure a Redux store
- Add a slice of state for books
- Add  reducers to add and remove book
- Add a slice of state for categories
- Add reducer in categoriesSlice that checks the status.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: {
+    book: [],
+  },
+  reducers: {
+    addbook: () => {},
+    removebook: () => {},
+  },
+});
```
Comment (ishimwezachee) [inline src/redux/books/booksSlice.js]: - [x]  You have done a great job add configure redux, but you still need to improve something on it. You have defined addBook and removeBook methods but they are empty. Please make sure that your methods contain the logic to remove and add a book as well.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi  @ankitt26, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image

## Highlights
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (ankitt26) [reply src/redux/books/booksSlice.js]: Thanks for reviewing, I am adding the logic.

**Review by abdallahmalima — APPROVED** [review]
Hi @ankitt26 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Setup react-redux✔️
- Use ducks pattern✔️
- Creates  `initialState`✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._✔️

**Comment by ankitt26** [comment]
Thanks for reviewing 🎉.

**Pull request merged**